src/Microsoft.AspNet.TestHost/TestServer.cs

```diff
@@ -129,7 +127,7 @@ public Task Invoke(HttpContext context)
             {
                 throw new ObjectDisposedException(GetType().FullName);
             }
-            return _appDelegate(context);
+            return _requestHandler(context.Features);
```

Comment:
Kind of a side effect of having the client code pass in an httpcontext, and the app invocation mediated by any TContext that can be created from a feature collection. Could be an interesting way for a different hosting.dll to plug a server into an owin-dictionary TContext for example.

It also helps re-enforce that the httpcontext itself is just a facade over the feature collection. Any httpcontext that stored primary data in a way other than as feature state wouldn't work through the test server.